### PR TITLE
Add typings for procmem module

### PR DESCRIPTION
**Pull request by psrok1**

### CODE REVIEW

File: malduck/procmem/procmem.py [hunk 1]
```diff
@@ -91,26 +129,42 @@ def __enter__(self):
...
+        memory = (
+            cast(bytearray, self.mapped_memory)
+            if self.mapped_memory is not None
+            else self.memory
+        )
+        if memory is None:
+            raise RuntimeError("ProcessMemory object is invalidated")
+        return memory
+
+    def close(self, copy: bool = False):
```
Comment (nazywam) [inline malduck/procmem/procmem.py:143]:
```suggestion
    def close(self, copy: bool = False) -> None:
```

File: malduck/procmem/procmem.py [hunk 2]
```diff
@@ -463,62 +538,78 @@ def patchv(self, addr, buf):
         region = self.addr_region(addr)
         # Boundary check
         if region is None or region.end < (addr + len(buf)):
-            raise ValueError("Cross-region patching is not supported")
+            raise ValueError("Address is out of single region")
```
Comment (nazywam) [inline malduck/procmem/procmem.py:541]: Maybe something like `Patched range spans across multiple regions`?

Comment (nazywam) [inline malduck/procmem/procmem.py:541]:
```suggestion
            raise ValueError("Patched range spans across multiple regions")
```

File: malduck/procmem/procmem.py [hunk 3]
```diff
@@ -599,10 +710,14 @@ def regexp(self, query, offset=0, length=None):
         for entry in re.finditer(query, chunk, re.DOTALL):
             yield offset + entry.start()
 
-    def regexv(self, query, addr=None, length=None):
+    def regexv(
+        self, query: int, addr: Optional[int] = None, length: Optional[int] = None
```
Comment (nazywam) [inline malduck/procmem/procmem.py:714]: `query: int`

:thinking:

File: malduck/procmem/procmem.py [hunk 4]
```diff
@@ -710,10 +841,17 @@ def _findbytes(self, yara_fn, query, addr, length):
         else:
             return []
 
-    def findbytesp(self, query, offset=0, length=None):
+    def findbytesp(
+        self,
+        query: Union[str, bytes],
```
Comment (nazywam) [inline malduck/procmem/procmem.py:846]: Probably `query: bytes` is intended here?
There is probably some call somewhere to `findbytesp` with str as an argument

File: malduck/procmem/procmem.py [hunk 5]
```diff
@@ -728,10 +866,17 @@ def findbytesp(self, query, offset=0, length=None):
         """
         return iter(self._findbytes(self.yarap, query, offset, length))
 
-    def findbytesv(self, query, addr=None, length=None):
+    def findbytesv(
+        self,
+        query: Union[str, bytes],
```
Comment (nazywam) [inline malduck/procmem/procmem.py:871]: Same as above

File: malduck/procmem/procmemelf.py [hunk 6]
```diff
@@ -20,22 +23,31 @@ class ProcessMemoryELF(ProcessMemoryBinary):
...
+        detect_image: bool = False,
+    ) -> None:
         self._elf = None
-        super(ProcessMemoryELF, self).__init__(
+        super().__init__(
             buf, base=base, regions=regions, image=image, detect_image=detect_image
         )
 
-    def _elf_direct_load(self):
+    def _elf_direct_load(self) -> elftools.elf.elffile.ELFFile:
```
Comment (nazywam) [inline malduck/procmem/procmemelf.py:39]: :tarnation:

File: malduck/procmem/procmem.py (hunk 2 above)
Comment (psrok1) [reply malduck/procmem/procmem.py:541]: Actually it's not true, because range can be out of any regions.

File: malduck/procmem/procmem.py (hunk 4 above)
Comment (psrok1) [reply malduck/procmem/procmem.py:846]: Not sure. Query is usually provided as literal e.g. `p.findbytesp("22 33 44 55")` so can be both string and bytes.

File: malduck/procmem/procmem.py (hunk 3 above)
Comment (psrok1) [reply malduck/procmem/procmem.py]: :cpanie: fixed

**Pull request merged**